### PR TITLE
DRY, and helperize

### PR DESCRIPTION
**Pull request by zancas**
I broke a test.

### CODE REVIEW

**Comment by zancas** [comment]
Would it makes sense to defined new type:
`struct MemoBytes([u8; 511])`
?

It's first byte must always be a valid version number.
I think for v0 memo's we know some bytes at the end have to be `0u8`.

File: zingo-memo/src/memo_serde.rs [hunk 1]
```diff
@@ -98,11 +99,14 @@ pub fn parse_memo(memo: [u8; 511]) -> io::Result<ParsedMemo> {
                 transaction_heights_and_indexes,
             })
         }
-        _ => {
+        x @ _ => {
```
Comment (AloeareV) [inline zingo-memo/src/memo_serde.rs]: This can just be
```suggestion
        x => {
```

**Review by AloeareV — CHANGES_REQUESTED** [review]
Looks great, much DRYer! One small change

**Review by zancas — COMMENTED** [review]
OK, I made your suggested change!

**Review by AloeareV — APPROVED** [review]

**Pull request merged**